### DISCUSSION TITLE
How to check TFT lcd driver

### DISCUSSION THREAD

**Discussion by lepromer** [Bodmer/TFT_eSPI#1762] [General]
hello,
i bought Tftf display ILI9481, and it is not properly working, perhaps it has  other chip,
**_How to check TFT lcd driver._**
thank you

**Comment by Bodmer** [comment]
First run the Test and diagnostics example "TFT_ReadWrite_Test" to see if pixels can be read back from the CGRAM. If the sketch passes then there are registers that can be read that might help to identify the controlled. The ID register address varies between controllers.

**Reply by lepromer** [reply]
thank you @Bodmer for your reply, i uploaded  "TFT_ReadWrite_Test" to esp32, TFT screen worked and i saw purple color in TFT, serial terminal says  , "Pixel value read read error", and i tried other drivers ILI9481,9486,9488, i reduced SPI_READ_FREQUENCY and SPI_FREQUENCYbut it say still read error.

**Reply by Bodmer** [reply]
The colour is correct so your screen appears to be working for writes. What is not working properly? Post a picture of the back of the display and a link to the sellers website if you need more help.
If pixel colours cannot be read then neither will control register values.

Most sketches do not require reading the screen RAM.
